src/com/linkedin/parseq/TaskDescriptor.java

```diff
@@ -0,0 +1,23 @@
+package com.linkedin.parseq;
+
+/**
+ * An API to provide description for task.
+ *
+ * parseq-lambda-names provide an implementation for this interface using Java SPI. Any changes made to this interface
+ * might require a change to implementation in parseq-lambda-names.
+ *
+ * The implementation doesnt need to be thread-safe.
+ *
+ * @author Siddharth Sodhani (ssodhani@linkedin.com)
+ */
+public interface TaskDescriptor {
```

Comment:
I think it would be best to get rid of this interface and replace it with Function<String, String>.
The main disadvantage is that it requires dependency from parseq-lambda-names to parseq main artifact. Currently parseq artifact is included in parseq-lambda-names which is not a good idea because we might have two different versions of parseq main artifact coming from two different class loaders
If we change that and parseq-lambda-names will assume that parseq main artifact is available at runtime we have two other problems. First: what if it isn't? We might see some class loading exceptions. Second: class loading is very tricky and we have to make sure that parseq artifacts are reachable from parseq-lambda-names regardless of which class loaders they are coming from.
Having TaskDescriptor interface is elegant but IMO it is not worth the  additional complexity. If we get rid of this interface the parseq main artifact would be completely independent from parseq-lambda-names